### PR TITLE
Updated documentation with Azure central pipeline use case

### PR DESCRIPTION
**Pull request by lukelloyd1985**
Addresses #4146 

## Proposed Changes

Added to documentation to help with an Azure pipeline use case

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -609,6 +609,48 @@ Add the following job in your `azure-pipelines.yaml` file
           artifactName: MegaLinterReport
 ```
 
+If you want or need to have a central pipeline with a build validation branch policy covering all repositories then you need to make some modifications so the pipeline uses the correct git repository.
```
Comment (nvuillam) [inline README.md:612]: Plz can you add headings for both cases, like:

### Pipeline in the repository

### Shared pipeline

**Review by nvuillam — CHANGES_REQUESTED** [review]
plz check comment :)

File: README.md (hunk 1 above)
Comment (lukelloyd1985) [reply README.md]: added and done some re-working too

**Review by nvuillam — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lukelloyd1985) [reply README.md]: also updated ToC that I previously missed 🤦‍♀️

**Comment by lukelloyd1985** [comment]
@nvuillam, can we get this PR merged?

**Comment by nvuillam** [comment]
@lukelloyd1985 done :)

**Pull request merged**